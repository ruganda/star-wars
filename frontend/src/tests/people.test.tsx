import TestRenderer from "react-test-renderer";
import { MockedProvider } from "@apollo/client/testing";
import { peopleMocks } from "./mock";
import PeopleContextProvider from "../contexts/peopleContext";
import People from '../people'



describe("<People />", () => {
  it("renders the table component without crashing", () => {
    const component = TestRenderer.create(
      <MockedProvider mocks={peopleMocks} addTypename={false}>
        <PeopleContextProvider>
          <People />
        </PeopleContextProvider>
      </MockedProvider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});