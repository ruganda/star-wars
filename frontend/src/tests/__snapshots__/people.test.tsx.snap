// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<People /> renders the table component without crashing 1`] = `
Array [
  <form
    onSubmit={[Function]}
  >
    <div
      className="MuiFormControl-root MuiTextField-root"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
        data-shrink={false}
        htmlFor="outlined-search"
        id="outlined-search-label"
      >
        Search name
      </label>
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputTypeSearch"
          disabled={false}
          id="outlined-search"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="search"
          value=""
        />
        <fieldset
          aria-hidden={true}
          className="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="PrivateNotchedOutline-legendLabelled-4"
          >
            <span>
              Search name
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-searchButton-1"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="submit"
    >
      <span
        className="MuiButton-label"
      >
        Search
      </span>
    </button>
  </form>,
  <div
    className="MuiPaper-root MuiTableContainer-root makeStyles-rootContainer-7 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <table
      aria-label="a dense table"
      className="MuiTable-root makeStyles-table-6"
      role={null}
    >
      <thead
        className="MuiTableHead-root"
        role={null}
      >
        <tr
          className="MuiTableRow-root MuiTableRow-head"
          role={null}
        >
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
            scope="col"
          >
            Name
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
            scope="col"
          >
            Gender
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
            scope="col"
          >
            Height
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
            scope="col"
          >
            Mass
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
            scope="col"
          >
            Homeworld
          </th>
        </tr>
      </thead>
      <tbody
        className="MuiTableBody-root"
        role={null}
      >
        <tr
          className="MuiTableRow-root makeStyles-muiTableRow-8"
          role={null}
        >
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
            role="cell"
            scope="row"
          >
            <span
              className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style={
                Object {
                  "height": undefined,
                  "width": undefined,
                }
              }
            />
          </th>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          >
             
            <span
              className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style={
                Object {
                  "height": undefined,
                  "width": undefined,
                }
              }
            />
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          >
            <span
              className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style={
                Object {
                  "height": undefined,
                  "width": undefined,
                }
              }
            />
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          >
            <span
              className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style={
                Object {
                  "height": undefined,
                  "width": undefined,
                }
              }
            />
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          >
            <span
              className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style={
                Object {
                  "height": undefined,
                  "width": undefined,
                }
              }
            />
          </td>
        </tr>
        <tr
          className="MuiTableRow-root makeStyles-muiTableRow-8"
          role={null}
        >
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
            role="cell"
            scope="row"
          >
            <span
              className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style={
                Object {
                  "height": undefined,
                  "width": undefined,
                }
              }
            />
          </th>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          >
             
            <span
              className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style={
                Object {
                  "height": undefined,
                  "width": undefined,
                }
              }
            />
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          >
            <span
              className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style={
                Object {
                  "height": undefined,
                  "width": undefined,
                }
              }
            />
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          >
            <span
              className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style={
                Object {
                  "height": undefined,
                  "width": undefined,
                }
              }
            />
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          >
            <span
              className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style={
                Object {
                  "height": undefined,
                  "width": undefined,
                }
              }
            />
          </td>
        </tr>
        <tr
          className="MuiTableRow-root makeStyles-muiTableRow-8"
          role={null}
        >
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
            role="cell"
            scope="row"
          >
            <span
              className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style={
                Object {
                  "height": undefined,
                  "width": undefined,
                }
              }
            />
          </th>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          >
             
            <span
              className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style={
                Object {
                  "height": undefined,
                  "width": undefined,
                }
              }
            />
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          >
            <span
              className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style={
                Object {
                  "height": undefined,
                  "width": undefined,
                }
              }
            />
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          >
            <span
              className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style={
                Object {
                  "height": undefined,
                  "width": undefined,
                }
              }
            />
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          >
            <span
              className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style={
                Object {
                  "height": undefined,
                  "width": undefined,
                }
              }
            />
          </td>
        </tr>
        <tr
          className="MuiTableRow-root makeStyles-muiTableRow-8"
          role={null}
        >
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
            role="cell"
            scope="row"
          >
            <span
              className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style={
                Object {
                  "height": undefined,
                  "width": undefined,
                }
              }
            />
          </th>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          >
             
            <span
              className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style={
                Object {
                  "height": undefined,
                  "width": undefined,
                }
              }
            />
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          >
            <span
              className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style={
                Object {
                  "height": undefined,
                  "width": undefined,
                }
              }
            />
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          >
            <span
              className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style={
                Object {
                  "height": undefined,
                  "width": undefined,
                }
              }
            />
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          >
            <span
              className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style={
                Object {
                  "height": undefined,
                  "width": undefined,
                }
              }
            />
          </td>
        </tr>
        <tr
          className="MuiTableRow-root makeStyles-muiTableRow-8"
          role={null}
        >
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
            role="cell"
            scope="row"
          >
            <span
              className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style={
                Object {
                  "height": undefined,
                  "width": undefined,
                }
              }
            />
          </th>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          >
             
            <span
              className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style={
                Object {
                  "height": undefined,
                  "width": undefined,
                }
              }
            />
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          >
            <span
              className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style={
                Object {
                  "height": undefined,
                  "width": undefined,
                }
              }
            />
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          >
            <span
              className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style={
                Object {
                  "height": undefined,
                  "width": undefined,
                }
              }
            />
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          >
            <span
              className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style={
                Object {
                  "height": undefined,
                  "width": undefined,
                }
              }
            />
          </td>
        </tr>
        <tr
          className="MuiTableRow-root makeStyles-muiTableRow-8"
          role={null}
        >
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
            role="cell"
            scope="row"
          >
            <span
              className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style={
                Object {
                  "height": undefined,
                  "width": undefined,
                }
              }
            />
          </th>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          >
             
            <span
              className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style={
                Object {
                  "height": undefined,
                  "width": undefined,
                }
              }
            />
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          >
            <span
              className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style={
                Object {
                  "height": undefined,
                  "width": undefined,
                }
              }
            />
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          >
            <span
              className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style={
                Object {
                  "height": undefined,
                  "width": undefined,
                }
              }
            />
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          >
            <span
              className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style={
                Object {
                  "height": undefined,
                  "width": undefined,
                }
              }
            />
          </td>
        </tr>
        <tr
          className="MuiTableRow-root makeStyles-muiTableRow-8"
          role={null}
        >
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
            role="cell"
            scope="row"
          >
            <span
              className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style={
                Object {
                  "height": undefined,
                  "width": undefined,
                }
              }
            />
          </th>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          >
             
            <span
              className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style={
                Object {
                  "height": undefined,
                  "width": undefined,
                }
              }
            />
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          >
            <span
              className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style={
                Object {
                  "height": undefined,
                  "width": undefined,
                }
              }
            />
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          >
            <span
              className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style={
                Object {
                  "height": undefined,
                  "width": undefined,
                }
              }
            />
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          >
            <span
              className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style={
                Object {
                  "height": undefined,
                  "width": undefined,
                }
              }
            />
          </td>
        </tr>
        <tr
          className="MuiTableRow-root makeStyles-muiTableRow-8"
          role={null}
        >
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
            role="cell"
            scope="row"
          >
            <span
              className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style={
                Object {
                  "height": undefined,
                  "width": undefined,
                }
              }
            />
          </th>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          >
             
            <span
              className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style={
                Object {
                  "height": undefined,
                  "width": undefined,
                }
              }
            />
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          >
            <span
              className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style={
                Object {
                  "height": undefined,
                  "width": undefined,
                }
              }
            />
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          >
            <span
              className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style={
                Object {
                  "height": undefined,
                  "width": undefined,
                }
              }
            />
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          >
            <span
              className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style={
                Object {
                  "height": undefined,
                  "width": undefined,
                }
              }
            />
          </td>
        </tr>
        <tr
          className="MuiTableRow-root makeStyles-muiTableRow-8"
          role={null}
        >
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
            role="cell"
            scope="row"
          >
            <span
              className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style={
                Object {
                  "height": undefined,
                  "width": undefined,
                }
              }
            />
          </th>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          >
             
            <span
              className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style={
                Object {
                  "height": undefined,
                  "width": undefined,
                }
              }
            />
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          >
            <span
              className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style={
                Object {
                  "height": undefined,
                  "width": undefined,
                }
              }
            />
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          >
            <span
              className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style={
                Object {
                  "height": undefined,
                  "width": undefined,
                }
              }
            />
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          >
            <span
              className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style={
                Object {
                  "height": undefined,
                  "width": undefined,
                }
              }
            />
          </td>
        </tr>
        <tr
          className="MuiTableRow-root makeStyles-muiTableRow-8"
          role={null}
        >
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
            role="cell"
            scope="row"
          >
            <span
              className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style={
                Object {
                  "height": undefined,
                  "width": undefined,
                }
              }
            />
          </th>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          >
             
            <span
              className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style={
                Object {
                  "height": undefined,
                  "width": undefined,
                }
              }
            />
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          >
            <span
              className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style={
                Object {
                  "height": undefined,
                  "width": undefined,
                }
              }
            />
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          >
            <span
              className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style={
                Object {
                  "height": undefined,
                  "width": undefined,
                }
              }
            />
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          >
            <span
              className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style={
                Object {
                  "height": undefined,
                  "width": undefined,
                }
              }
            />
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
]
`;
